connectors/rabbitmq/src/main/java/io/camunda/connector/rabbitmq/inbound/RabbitMqInboundConnectorSubscription.java

```diff
@@ -0,0 +1,84 @@
+ */
+package io.camunda.connector.rabbitmq.inbound;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.DeliverCallback;
+import io.camunda.connector.api.annotation.InboundConnector;
+import io.camunda.connector.api.inbound.InboundConnectorContext;
+import io.camunda.connector.api.inbound.InboundConnectorProperties;
+import io.camunda.connector.api.inbound.SubscriptionInboundConnector;
+import io.camunda.zeebe.client.api.response.ProcessInstanceEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+@InboundConnector(type = "rabbitmq", name = "RABBIT_MQ")
+public class RabbitMqInboundConnectorSubscription implements SubscriptionInboundConnector {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RabbitMqInboundConnectorSubscription.class);
+
+  public RabbitMqInboundConnectorSubscription() {
+  }
+
+  @Override
+  public void activate(InboundConnectorProperties genericProperties, final InboundConnectorContext context) throws Exception {
+    // TODO: What happens if this fails because of some error of the third party - when would we retry connecting?
+    // Would that be with the next polling interval as this is not marked as active? Double check!
+    final RabbitMqConnectorProperties properties = new RabbitMqConnectorProperties(genericProperties);
+
+    ConnectionFactory connectionFactory = new ConnectionFactory();
+    connectionFactory.setUri(properties.getUri());
+    Connection connection = connectionFactory.newConnection();
+    Channel channel = connection.createChannel();
+
+    DeliverCallback deliverCallback = (consumerTag, delivery) -> {
+      HashMap<String, Object> message = new HashMap<>();
+      message.put("envelope", delivery.getEnvelope());
+      message.put("properties", delivery.getProperties());
+      message.put("body", new String(delivery.getBody(), StandardCharsets.UTF_8));
+
+      executeOnIncomingMessage(properties, context, message);
+      LOG.trace("Received AMQP message %1", message);
+    };
+
+    channel.basicConsume(properties.getQueueName(), deliverCallback, consumerTag -> { /*Canel Callback*/ });
+  }
+
+  /**
+   * Copied from InboundWebhookController - maybe better make some code reusable in connector-runtime-util
+   */
+  private ProcessInstanceEvent executeOnIncomingMessage(RabbitMqConnectorProperties connectorProperties, InboundConnectorContext context, Map<String, Object> message) {
+    final Map<String, Object> variables = extractVariables(connectorProperties, message);
+
+    return context.zeebeClient()
+            .newCreateInstanceCommand()
+            .bpmnProcessId(connectorProperties.getBpmnProcessId())
+            .version(connectorProperties.getVersion())
+            .variables(variables)
+            .send()
+            .join();
+  }
+
```

Comment:
Hey @berndruecker I like the general approach!

My 2 cents:
1. I would prefer the concrete connector implementations to be abstracted away from communication with Zeebe and process-specific details like `processId`, `version` and other potential target process identifiers.
So if we follow this idea, the connectors would ideally just call a pre-defined method when the inbound event occurs, and the SDK/runtime code will do the rest of the job correlating the message.

2. Why don't we use process messages for all inbound correlations? We can start a process with a message, we can correlate it to an already running process, so this would be uniform for all inbound targets.

Reply:
I wonder if we already feel confident enough that an abstraction does not hinder us from doing what we need to? 
At the same time, it might be valuable that all inbound connectors follow the same schema when it comes to selecting the right process/version and/or message (like for example there was some logic in the webhook to clarify what happens if a second process version with the same/other context was deployed: https://github.com/camunda-community-hub/spring-zeebe/blob/master/connector-runtime/src/main/java/io/camunda/connector/runtime/inbound/registry/InboundConnectorRegistry.java#L92). 

For 2: As written in https://github.com/camunda/team-connectors/issues/305#issuecomment-1378750080 I think supporting messages only is a bit more challenging, especially as no exception is provided if a message cannot be correlated. At the same time we cannot influence the version of a process instance started - which might be a good or a bad thing - depending on the viewpoint :-)

Reply:
I agree, we shouldn't really overengineer the abstractions so early on 😄 
But one way or another, I'm trying to imagine how it fits together with using inbound connectors as intermediate events. Some sort of abstraction will be necessary, so that the connector code can work against both types of events.

Regarding 2, we'll still have to face these challenges to some extent while implementing the intermediate events, right? But let's not rush with this idea, I'm sure we can generalize the approach later on if it starts to make sense.